### PR TITLE
Load ini modules via ini section

### LINKED ISSUE #139: Allow loading external modules in the ini file via a full section
Currently external modules are loaded like:

[Game]
```
ExternalModules = "NetFix, NetHelper"
```

I wold prefer loading ExternalModules:
```
[ExternalModules]
NetFix = yes
NetHelper = no
```

However, I haven't researched Ini file access enough to know if this is supportable by the winapi. We can get away with this sort of list for BuiltInModules because we know all their names ahead of time. For an external module, we don't know the module's name before compilation.

### PR DESCRIPTION
**Pull request by Brett208**
Closes #139.
Partially addresses #76.

The different tests for ModuleLoader and modules are spread across 3 test files now. I think it is good we are separating them out, but maybe some thought should be placed into which tests are run in which file to prevent duplicating work and easing organization. Maybe better for a separate branch?

### CODE REVIEW

File: srcStatic/ModuleLoader.cpp [hunk 1]
```diff
@@ -67,22 +67,23 @@ void ModuleLoader::RegisterConsoleModules()
 
 void ModuleLoader::RegisterIniModules()
 {
-	const auto sectionNames = GetModuleNames("ExternalModules");
-
-	for (const auto& sectionName : sectionNames)
+	for (const auto moduleName : iniFile.GetKeyNames("ExternalModules"))
```
Comment (Brett208) [inline srcStatic/ModuleLoader.cpp]: I waffled between ExternalModules and IniModules. Sort of weird to have an IniModules section name in an ini file. Happy to change if @DanRStevens has preference one way or other?

File: test/IniModule.test.cpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+		//iniFileStream << "Test = yes\n";
+	}
+
+	iniFileStream.close();
+}
+
+TEST(IniModule, NoDll)
+{
+	const auto iniFilename = fs::path(GetExeDirectory()) / fs::path("IniModuleTest.ini");
+	WriteExternalModuleIniFile(iniFilename.string(), { "Test = yes" });
```
Comment (Brett208) [inline test/IniModule.test.cpp]: I'm not sure it is legal to pass into a string_view argument into a function without first recording the string to variable. Seems to work fine on MSVC though.

File: test/IniModule.test.cpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+void WriteExternalModuleIniFile(std::string_view iniFilename, const std::vector<std::string_view> sectionPairs)
+{
+	std::ofstream iniFileStream(iniFilename);
+
+	iniFileStream << "[ExternalModules]\n";
+
+	for (const auto& sectionPair : sectionPairs)
+	{
+		iniFileStream << sectionPair << "\n";
+		//iniFileStream << "Test = yes\n";
```
Comment (DanRStevens) [inline test/IniModule.test.cpp]: Maybe get rid of this commented out line.

File: test/IniModule.test.cpp (hunk 2 above)
Comment (DanRStevens) [reply test/IniModule.test.cpp]: Looks like the standard library packaged with Mingw doesn't have an overload taking `std::string_view` for the `std::ofstream` constructor. I guess we can just use a `const std::string&` for now.

File: srcStatic/ModuleLoader.cpp (hunk 1 above)
Comment (DanRStevens) [reply srcStatic/ModuleLoader.cpp]: Uhh, I don't feel joy about these names, but I don't have a better suggestion right now. Maybe best to look at this again when I'm not tired. I'll open an issue to track.

File: test/IniModule.test.cpp (hunk 3 above)
Comment (Brett208) [reply test/IniModule.test.cpp]: Thanks for catching my sloppiness here.

File: test/IniModule.test.cpp (hunk 2 above)
Comment (Brett208) [reply test/IniModule.test.cpp]: Okay, the other option would be to define our own overload of ofstream for mingw that takes a string_view. :-) I know how to do that in C#, not sure what the C++ equivalent is without researching (I assume it is possible).

Thanks for fixing for me.

**Comment by Brett208** [comment]
Thanks for the quick review. Yeah, it actually shortened the non unit test code base a bit and makes the ini file nicer to work with. The unit tests are lacking both because there is no way to test normal flow (no errors) and there is no real way to tell if the right errors are encountered in the 'bad' test cases. As mentioned, probably best for a different branch to deal with.

I made a slight improvement using std:endl after approval, but didn't think this was worth another review.

-Brett

File: test/IniModule.test.cpp (hunk 2 above)
Comment (DanRStevens) [reply test/IniModule.test.cpp]: We would have to add/modify things in the `std::` namespace, which creates upgrade and portability issues. Probably best to just leave this one alone.

It's also likely we'd need to do a `std::string_view` to `std::string` conversion to get it working, which kind of defeats the purpose of using `std::string_view` in the first place.

**Pull request merged**